pxtblocks/code-validation/rubric.ts

```diff
@@ -0,0 +1,30 @@
+namespace pxt.blocks {
+    export class Rubric {
+        criteria: RubricCriteria[];
+
+        constructor(criteria: RubricCriteria[]) {
+            this.criteria = criteria;
+        }
+    }
+
+    export function parseRubric(rubric: string): Rubric {
+        let rubricObj;
+        try {
+            rubricObj = JSON.parse(rubric);
+        } catch (e) {
+            console.error(`Error parsing rubric! ${e}`);
+            return null;
+        }
+
+        if (!rubricObj.criteria) {
+            console.error(`No criteria found in rubric`);
+            return null;
+        }
+
+        const criteria: RubricCriteria[] = rubricObj.criteria.map((c: CriteriaData) => {
+            return getCriteria(c);
+        }).filter((r: RubricCriteria) => !!r);
```

Comment:
Just wanting to clarify, this filter is here to get rid of any entries that couldn't be processed? 

This might be something that we want to talk about.. I would generally like the sort of system that doesn't fail early if there are entries that are formatted incorrectly, but that might be somewhat frustrating for the rubric case. If a teacher submitted a rubric to our tool with a criteria that isn't defined or formatted correctly, and then they ran that rubric against a student file and only got results from the criteria that we could accept, would that be more work for them? Would they want those results from the few that worked or would it be better to have them re-submit the rubric where everything is formatted correctly?

I don't think there's anything to change right now, but it's just something to keep in mind. This will likely be a more UI-centered solution anyway. Maybe having something where we inform the user that some of the criteria in the rubric couldn't be parsed and ask them if they would like to run the rubric or if they would want to reformat the rubric and resubmit or something like that.

Reply:
Yea, all good points. My thought process was that we would be formatting the rubric behind the scenes based on the teacher's selections in the UI - so there shouldn't be any rubric criteria that is ill-formatted. But yes, in the future we would want some kind of indication if any of the rubric criteria wasn't successfully parsed.